python/tk_desktop/systray.py

```diff
@@ -332,25 +325,21 @@ def activate(self):
             self.show()
             self.raise_()
             self.activateWindow()
-            if osutils is not None:
-                osutils.make_app_foreground()
+            osutils.make_app_foreground()
         else:
             # shown and not topmost, just bring to the top
             self.raise_()
             self.activateWindow()
-            if osutils is not None:
-                osutils.activate_application()
+            osutils.activate_application()
 
     def toggle_activate(self):
         """
         Toggles visibility when systray icon is clicked.
         """
-        active = self.isActiveWindow()
-        if active:
+        if self.isVisible():
```

Comment:
This is the actual fix for the pinned dialog. In PySide2, `isActiveWindow` always returns true for some reason. Using isVisible makes sense and works for both PySide and PySide2, so we're good now for all versions of Desktop.